gtsfm/common/gtsfm_data.py

```diff
@@ -351,14 +354,14 @@ def filter_landmarks(self, reproj_err_thresh: float = 5) -> "GtsfmData":
         # TODO: move this function to utils or GTSAM
         filtered_data = GtsfmData(self.number_images())
 
-        # add all the cameras
-        for i in self.get_valid_camera_indices():
-            filtered_data.add_camera(i, self.get_camera(i))
-
         for j in range(self.number_tracks()):
             track = self.get_track(j)
 
             if self.__validate_track(track, reproj_err_thresh):
+                # check if all cameras with measurement in this track have already been added
+                for k in range(track.number_measurements()):
+                    i, _ = track.measurement(k)
+                    filtered_data.add_camera(i, self.get_camera(i))
```

Comment:
thanks for the PR @codyly . Won't this add the same camera over and over again? shouldn't we add just once?

Reply:
I also modified the `add_camera` function to check whether the same camera has been added

Reply:
If we do not modify the `add_camera`, it can be
```py
    for k in range(track.number_measurements()):
        i, _ = track.measurement(k)
        if i not in self._cameras:
            filtered_data.add_camera(i, self.get_camera(i))
```

Reply:
Ah I see, ok what you have already works for me then.